prime-dotnet-webapi/Services/SiteService.cs

```diff
@@ -332,6 +332,26 @@ public async Task DeleteSiteAsync(int siteId)
             }
         }
 
+        public async Task<Site> ApproveSite(int siteId)
+        {
+            var site = await this.GetBaseSiteQuery()
+                .SingleOrDefaultAsync(s => s.Id == siteId);
+
+            if (site != null && site.ApprovedDate == null)
+            {
+                site.ApprovedDate = DateTimeOffset.Now;
+
+                this._context.Update(site);
```

Comment:
shouldn't need this line

Reply:
Why is that? Just going off other methods that update that have this line. As well as not having it there would make me believe context would not know of the update. Unless the "site" when we get it from the search is a reference to the original object and the context tracks changes to that object?

Reply:
When you pull an object from the database, entity framework tracks changes you make to the object. It already knows the object is updated. The few places we use context.Update in the code base are generally because we have a situation where a child object is being either added, modified, or removed because of the update and we want to make it explicit. There are also a few uses of context.update that are also not needed and we could remove them if we wanted to.

Reply:
se, for example, in the Submissionservice:
`public async Task UpdateAlwaysManualAsync(int enrolleeId, bool alwaysManual)
        {
            var enrollee = await _context.Enrollees
                .SingleAsync(e => e.Id == enrolleeId);

            enrollee.AlwaysManual = alwaysManual;
            await _context.SaveChangesAsync();
        }`